src/app/_components/RecentBlocks/useRecentBlocks.ts

```diff
@@ -0,0 +1,131 @@
+  return block.height > (lastBlockHeight || 0);
+};
+
+const isNewBtcBlock = (block: NakamotoBlock | Block, lastBlockHeight: number | undefined) => {
+  return block.burn_block_height > (lastBlockHeight || 0);
+};
+
+// Handle getting stx blocks data. If there is initial stx blocks data, set it in the cache and use it. Otherwise, query for the data
+const useStxBlocks = () => {
+  const { initialRecentBlocks } = useHomePageData();
+  const activeNetwork = useGlobalContext().activeNetwork;
+
+  const queryClient = useQueryClient();
+
+  const recentStxBlocks = initialRecentBlocks?.stxBlocks;
+  const isStxBlocksCacheSetWithInitialData = useRef(false);
+  if (isStxBlocksCacheSetWithInitialData.current === false && recentStxBlocks) {
+    const queryKey = getBlocksV2ListQueryKey(RECENT_STX_BLOCKS_COUNT, activeNetwork.url);
+    queryClient.setQueryData(queryKey, recentStxBlocks);
+    isStxBlocksCacheSetWithInitialData.current = true;
+  }
+  const { data: stxBlocksData, refetch: stxRefetch } = useBlocksV2List(RECENT_STX_BLOCKS_COUNT, {
+    enabled: false,
+  });
+  const stxBlocks = stxBlocksData?.results || [];
+  return { stxBlocks, stxRefetch };
+};
+
+// Handle getting btc blocks data. If there is initial btc blocks data, set it in the cache and use it. Otherwise, query for the data
+const useBtcBlocks = () => {
+  const { initialRecentBlocks } = useHomePageData();
+  const activeNetwork = useGlobalContext().activeNetwork;
+
+  const queryClient = useQueryClient();
+
+  const recentBtcBlocks = initialRecentBlocks?.btcBlocks;
+  const isBtcBlocksCacheSetWithInitialData = useRef(false);
+  if (isBtcBlocksCacheSetWithInitialData.current === false && recentBtcBlocks) {
+    const queryKey = getBurnBlocksQueryKey({
+      limit: RECENT_BTC_BLOCKS_COUNT,
+      networkUrl: activeNetwork.url,
+      queryKeyExtension: 'recent',
+    });
+    queryClient.setQueryData(queryKey, recentBtcBlocks);
+    isBtcBlocksCacheSetWithInitialData.current = true;
+  }
+  const { data: btcBlocksData, refetch: btcRefetch } = useBurnBlocks(
+    RECENT_BTC_BLOCKS_COUNT,
+    undefined,
+    {
+      enabled: false,
+    }
+  );
+  const btcBlocks = btcBlocksData?.results || [];
+  return { btcBlocks, btcRefetch };
+};
+
+export function useRecentBlocks(recentBlocksType: RecentBlocksType) {
+  const [hasNewBtcBlocks, setHasNewBtcBlocks] = useState(false);
+  const [hasNewStxBlocks, setHasNewStxBlocks] = useState(false);
+  const [socketEnabled, setSocketEnabled] = useState(true);
+
+  const { stxBlocks, stxRefetch } = useStxBlocks();
+  const newestStxBlockHeight = stxBlocks[0]?.height;
+
+  const { btcBlocks, btcRefetch } = useBtcBlocks();
+  const newestBtcBlockHeight = btcBlocks[0]?.burn_block_height;
+
+  // Update socketEnabled state when we have new blocks for the selected type
+  useEffect(() => {
+    if (recentBlocksType === 'btc' && hasNewBtcBlocks && socketEnabled) {
+      setSocketEnabled(false);
```

Comment:
code disables socket when new block arrive, if the user then switches to the other tab without clicking update, does socket stay disabled AKA other tab won’t listen for new blocks?